Update console-virtual-terminal-sequences.md with Cursor Shape (#261)
MicrosoftDocs/Console-Docs: docs/console-virtual-terminal-sequences.md

| ESC \[ &lt;n&gt; A | CUU | Cursor Up | Cursor up by &lt;n&gt; |
| ESC \[ &lt;n&gt; B | CUD | Cursor Down | Cursor down by &lt;n&gt; |
| ESC \[ &lt;n&gt; C | CUF | Cursor Forward | Cursor forward (Right) by &lt;n&gt; |

> | ESC \[ &lt;n&gt; D | IND | Cursor Down | Cursor down one line from current position in the same column |

Reviewer: I've updated the notes here: https://github.com/MicrosoftDocs/Console-Docs/issues/94#issuecomment-1117609251

We do not need to make this change, because `ESC [ D` is correct (`CUB`)

Author: Reverted

> [!TIP]
> The enable sequences end in a lowercase H character (`h`) and the disable sequences end in a lowercase L character (`l`).

## <span id="Cursor_Shape"></span><span id="cursor_shape"></span><span id="CURSOR_SHAPE"></span>Cursor Shape

The following commands control and allow for customization of the cursor shape.

| Sequence | Code | Description | Behavior |
|---------------|---------|------------------------------|---------------------------|

> | ESC \[ 0 SP q | DECSCUSR | User Shape | Default cursor shape configured by the user |

Reviewer 2: Elsewhere on this page, the Description column shows how the control sequence is defined by the industry, and the Behavior column shows how Windows Terminal implements it. Did DEC define DECSCUSR with parameter 0 as activating the cursor shape preferred by the user, or is that specific to Windows Terminal?

Page 5-127 of <http://boundlessterminals.com/manuals/VT520-VT525ProgrammerInformation.pdf> lists "0, 1 or none" as selecting the "Blink Block (Default)" cursor style. That suggests to me that the number 1 means this style specifically, and 0 or none means the default style, which in those DEC terminals is hardcoded rather than a user preference. So the pull request looks OK in that respect.

Does Windows Terminal support DECSCUSR without a parameter? Perhaps that too should be documented here, something like the paragraph in another section of this page: <https://github.com/MicrosoftDocs/Console-Docs/blob/e1d1841c175e4f5b3c2f469d8515a975c83a28d9/docs/console-virtual-terminal-sequences.md?plain=1#L171>

| ESC \[ 1 SP q | DECSCUSR | Blinking Block | Blinking block cursor shape |
| ESC \[ 2 SP q | DECSCUSR | Steady Block | Steady block cursor shape |
| ESC \[ 3 SP q | DECSCUSR | Blinking Underline | Blinking underline cursor shape |
| ESC \[ 4 SP q | DECSCUSR | Steady Underline | Steady underline cursor shape |
| ESC \[ 5 SP q | DECSCUSR | Blinking Bar | Blinking bar cursor shape |
| ESC \[ 6 SP q | DECSCUSR | Steady Bar | Steady bar cursor shape |

> [!NOTE]

> > `SP` is a literal space character of 0x20 as an Intermediate Byte, followed by `q` which is 0x71 as a Final Byte. 

Reviewer: ```suggestion
> `SP` is a literal space character (0x20) in the intermediate position, and it is followed by `q` (0x71) in the final position. 
```

I slightly reworded what @KalleOlaviNiemitalo said in the linked issue; feel free to accept or reject this change. :)
